File: back-end/src/main/java/com/example/backend/dto/vehicle/VehicleDTO.java
package com.example.backend.dto.vehicle;

import com.example.backend.dto.base.BaseDTO;
import com.example.backend.entities.Vehicle;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class VehicleDTO extends BaseDTO {
    private String licensePlate;
    private String make;
    private String model;
    private Integer year;
    private Integer capacity;
    private LocalDate insuranceExpiryDate;
    private LocalDate registrationExpiryDate;
    private LocalDate lastMaintenanceDate;
    private Double currentMileage;
    private Long driverId;
    private String driverName;
    private Long schoolId;
    private String schoolName;
    private Double currentLatitude;
    private Double currentLongitude;
    private Vehicle.VehicleStatus status;
    private Long completedTripsCount;
    private Double totalMileage;
    private Long activeTripsCount;
} 

File: front-end/src/core/dto/vehicle.dto.ts
import type { BaseDTO } from "./base.dto"
import { VehicleStatus } from "../entities/vehicle.entity"

export interface VehicleDTO extends BaseDTO {
  vehicleNumber: string
  make: string
  model: string
  year: number
  capacity: number
  status: VehicleStatus
  currentMileage: number
  insuranceExpiryDate: string
  registrationExpiryDate: string
  lastMaintenanceDate: string
  trackingDeviceId?: string
  currentLatitude?: number
  currentLongitude?: number
}

export interface CreateVehicleRequest {
  vehicleNumber: string
  make: string
  model: string
  year: number
  capacity: number
  status: VehicleStatus
  currentMileage: number
  insuranceExpiryDate: string
  registrationExpiryDate: string
  lastMaintenanceDate: string
  trackingDeviceId?: string
  currentLatitude?: number
  currentLongitude?: number
  initialLatitude?: number
  initialLongitude?: number
  schoolId?: number
}

export interface UpdateVehicleRequest {
  vehicleNumber?: string
  make?: string
  model?: string
  year?: number
  capacity?: number
  status?: VehicleStatus
  currentMileage?: number
  insuranceExpiryDate?: string
  registrationExpiryDate?: string
  lastMaintenanceDate?: string
  trackingDeviceId?: string
  currentLatitude?: number
  currentLongitude?: number
  initialLatitude?: number
  initialLongitude?: number
} 